src/Hare/AppView.cpp

```diff
@@ -1125,3 +1156,50 @@ AppView::AlertUser(const char* message)
 							   B_WARNING_ALERT);
 	alert->Go();
 }
+
+
+BString
+AppView::ReplaceInvalidFileCharsInBString(BString filestr, int32 swaptype)
+{
+	// Swap types
+	// 1) ASCII to Dash
+	// 2) ASCII to Underscore
+	// 3) ASCII to UTF8 close match
+	switch (swaptype)
+	{
+		case 1:
+			filestr.ReplaceAllChars(SLASH, DASH, 0);
+			filestr.ReplaceAllChars(BACKSLASH, DASH, 0);
+			filestr.ReplaceAllChars(ASTERISK, DASH, 0);
+			filestr.ReplaceAllChars(QUOTE, DASH, 0);
+			filestr.ReplaceAllChars(COLON, DASH, 0);
+			filestr.ReplaceAllChars(QUESTIONMARK, DASH, 0);
+			filestr.ReplaceAllChars(GREATERTHAN, DASH, 0);
+			filestr.ReplaceAllChars(LESSTHAN, DASH, 0);
+			filestr.ReplaceAllChars(PIPE, DASH, 0);
+			break;
+		case 2:
+			filestr.ReplaceAllChars(SLASH,UNDERSCORE, 0);
+			filestr.ReplaceAllChars(BACKSLASH,UNDERSCORE, 0);
+			filestr.ReplaceAllChars(ASTERISK, UNDERSCORE, 0);
+			filestr.ReplaceAllChars(QUOTE,UNDERSCORE, 0);
+			filestr.ReplaceAllChars(COLON,UNDERSCORE, 0);
+			filestr.ReplaceAllChars(QUESTIONMARK,UNDERSCORE, 0);
+			filestr.ReplaceAllChars(GREATERTHAN,UNDERSCORE, 0);
+			filestr.ReplaceAllChars(LESSTHAN,UNDERSCORE, 0);
+			filestr.ReplaceAllChars(PIPE,UNDERSCORE, 0);
+			break;
```

Comment:
case 1 +2 coud use BString's CharacterEscape()

Maybe rename to ReplaceInvalidFileChars() to make it less of a mouthfull. :)

Reply:
I just tried out CharacterEscape().  It inserts a the given char before each of the ones I'd like to replace, so seems it won't work for this.  I also tried out ReplaceCharsSet figuring it would take a setOfChars and if any of those chars were found it would replace them.  Nope.  I think it requires it to match all of the setOfChars, so another no-go.
I did shorten the function name as suggested.

Reply:
Oh, I should have investigated first... Sorry to have sent you on a wrong path.